lib/internal/net.js

```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = { isLegalPort };
+
+// Check that the port number is not NaN when coerced to a number,
+// is an integer and that it falls within the legal range of port numbers.
+function isLegalPort(port) {
+  if (typeof port === 'string' && port.trim() === '')
+    return false;
+  return +port === (port >>> 0) && port >= 0 && port <= 0xFFFF;
```

Comment:
The problem with `parseInt()` is that something like `'10c'` becomes 10, which is not right. The problem with `Number.isInteger()` is that `'0xff'` is false. There are all kinds of weird edge cases.